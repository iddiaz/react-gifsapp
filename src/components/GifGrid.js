// import React, {useState, useEffect } from 'react'
import React from 'react'
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from './../hooks/useFetchGifs';

export const GifGrid = ({category}) => {


   const {data:images , loading } = useFetchGifs( category );




   return (
      <>
         <h3> {category} </h3>
         <div className="card-grid">

            { loading && <p className="animate__animated animate__flash">Loading</p>  }
            
               {
                  images.map(( img) => (
                     <GifGridItem 
                        key={img.id} 
                        // img={img} 
                        {...img}
                        />
                  ))
               }            
         </div>      
      </>
   )
}
