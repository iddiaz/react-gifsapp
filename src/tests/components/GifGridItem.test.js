import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"
import { React } from 'react';
import '@testing-library/jest-dom'

describe('Prubeas en <GifGridItem />', ()=> {

   const title = 'Un titulo';
   const url = 'http://localhost/algo.jpg'

   const wrapper = shallow( <GifGridItem title={title} url={url} /> )

   test('Debe mostrar el componente corretamente', () => {

      expect( wrapper ).toMatchSnapshot();      
      
   })


   test('Debe tener un pÃ¡rrafo con el titulo', () => {

      const p = wrapper.find('p');

      expect (p.text().trim()).toBe(title);
      
   })

   test('Debe tener ula imagen igual al url y al de props', () => {

      const img = wrapper.find('img');

      // console.log(img.html());
      // console.log(img.props()); 
      // console.log(img.prop('src'));
      
      expect( img.prop('src') ).toBe( url );
      expect( img.prop('alt') ).toBe( title );
      
   })

   test('Debe tener animate__bounce', () => {

      const div = wrapper.find('div');  
      
      console.log(div.props()); 

      const className = div.prop('className');
      
      expect( className.includes('animate__bounce') ).toBe(true) 

      
   })


})