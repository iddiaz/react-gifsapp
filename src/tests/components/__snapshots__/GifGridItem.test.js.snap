// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubeas en <GifGridItem /> Debe mostrar el componente corretamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
     
  </p>
</div>
`;
